File: proga/1_semestr/labs/lab3/src/main/java/dev/lysmux/lab3/entity/place/Table.java
package dev.lysmux.lab3.entity.place;

import dev.lysmux.lab3.common.Case;
import dev.lysmux.lab3.entity.item.MoonStone;

import java.util.Objects;

public class Table extends Place {
    @Override
    public String caseDeclension(Case to_case) {
        return switch (to_case) {
            case NOMINATIVE, ACCUSATIVE -> "стол";
            case GENITIVE -> "стола";
            case DATIVE -> "столу";
            case CREATIVE -> "столом";
            case PREPOSITIONAL -> "столе";
        };
    }

    @Override
    public String toString() {
        return "стол";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Table table)) return false;
        if (!super.equals(o)) return false;
        return Objects.equals(table.toString(), table.toString());
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), toString());
    }
}

File: proga/1_semestr/labs/lab3/src/main/kotlin/dev/lysmux/lab3/entity/place/Table.kt
package dev.lysmux.lab3.entity.place

import dev.lysmux.lab3.common.Case

class Table : Place() {
    override fun caseDeclension(toCase: Case): String {
        return when (toCase) {
            Case.NOMINATIVE -> "стол"
            Case.GENITIVE -> "стола"
            Case.DATIVE -> "столу"
            Case.ACCUSATIVE -> "стол"
            Case.CREATIVE -> "столом"
            Case.PREPOSITIONAL -> "столе"
        }
    }

    override fun toString() = "стол"

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        return other is Table
    }

    override fun hashCode() = javaClass.hashCode()
}